### PR TITLE
Bump tekton/pipeline to v0.27.1

### PR DESCRIPTION
**Pull request by piyush-garg**
This will bump tektoncd/pipeline to v0.27.1
Also bump k8s and knative deps which needs
changes in client set generation etc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
Bump tekton/pipeline to v0.27.1
```

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/meow

File: test/framework/clients.go [hunk 1]
```diff
@@ -45,15 +45,16 @@ func NewClients(configPath, clusterName, namespace string) *Clients {
...
 
-	c.KubeClient, err = knativetest.NewKubeClient(configPath, clusterName)
+	cfg, err := knativetest.BuildClientConfig(configPath, clusterName)
 	if err != nil {
-		log.Fatalf("failed to create kubeclient from config file at %s: %s", configPath, err)
+		log.Fatalf("failed to create configuration obj from %s for cluster %s: %s", configPath, clusterName, err)
 	}
 
-	cfg, err := knativetest.BuildClientConfig(configPath, clusterName)
+	kubeClient, err := kubernetes.NewForConfig(cfg)
```
Comment (vinamra28) [inline test/framework/clients.go:53]: can't we do `c.KubeClient, err = kubernetes.NewForConfig(cfg)` ?

**Review by vinamra28 — COMMENTED** [review]
just one nit otherwise looks good

File: test/framework/clients.go (hunk 1 above)
Comment (piyush-garg) [reply test/framework/clients.go:53]: we are setting it after checking the error

**Comment by vinamra28** [comment]
/lgtm
Thanks :+1:

**Pull request merged**